modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java

```diff
@@ -0,0 +1,65 @@
+package com.etendorx.das.utils;
+
+import lombok.extern.slf4j.Slf4j;
+import org.hibernate.QueryException;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
+public class DefaultFilters {
+    public static String addFilters(String sql, String adUser, String clientId, String orgId, String isActive) {
+
+        //SUPERUSER BY PASS FILTERS
+        if (isSuperUser(adUser, clientId, orgId)) return sql;
+
+        boolean containsWhere = sql.contains("where");
+        //GET THE TABLE NAME USING REGULAR EXPRESSION
+        String tableName = getTableName(sql);
+
+        //ADD DEFAULT FILTERS
+        String resultSql = replaceInQuery(sql, clientId, orgId, isActive, containsWhere, tableName);
+
+        return resultSql;
+    }
+
+    @NotNull
+    private static String replaceInQuery(String sql, String clientId, String orgId, String isActive, boolean containsWhere, String tableName) {
+        String resultSql;
+        if (containsWhere) {
+            resultSql = sql.replace("where", "where "+ tableName +".ad_client_id = '"+ clientId +"' " +
+                    "and "+ tableName +".ad_org_id = ANY (etrx_org_tree('"+ orgId +"')) " +
+                    "and "+ tableName +".isactive = '"+ isActive +"' and ");
+        } else {
+            resultSql = sql.replace("limit", "where "+ tableName +".ad_client_id = '"+ clientId +"' " +
+                    "and "+ tableName +".ad_org_id = ANY (etrx_org_tree('"+ orgId +"')) " +
+                    "and "+ tableName +".isactive = '"+ isActive +"' limit");
+        }
+        return resultSql;
+    }
+
+    private static boolean isSuperUser(String adUser, String clientId, String orgId) {
+        if (adUser.compareTo("100") == 0 && clientId.compareTo("0") == 0 && orgId.compareTo("0") == 0 ) {
+            return true;
+        }
+        return false;
+    }
+
+    private static String getTableName(String sql) {
+        Pattern selectPattern = Pattern.compile("select\\s+(\\w+)\\.");
```

Comment:
private static String getTableName(String sql) { 
        Pattern pattern = Pattern.compile("select\\s+(\\w+)\\.");
        Matcher matcher = pattern.matcher(sql);
        if (matcher.find()) {
            return matcher.group(1);
        } else {
            pattern = Pattern.compile("select\\s+count\\(\\s*(\\w+)\\.\\w+\\)\\s+as");
            matcher = pattern.matcher(sql);
            if (matcher.find()) {
                return matcher.group(1);
            }
        }
        log.error("[ getTableName ] - PATTERN ERROR");
        throw new QueryException("getTableName ERROR ");
  }

 Less vars and less patters's checks.

Reply:
getTableName method was refactorized to getTableAlias, and uses only 1 matching pattern